profiler/controllers.py

```diff
@@ -51,21 +52,31 @@ def dump(self, token, **post):
         Uses a temporary file, because apparently there's no API to
         dump stats in a stream directly.
         """
-        handle, path = mkstemp(prefix='profiling')
-        core.profile.dump_stats(path)
-        stream = os.fdopen(handle)
-        os.unlink(path)  # TODO POSIX only ?
-        stream.seek(0)
-        filename = 'openerp_%s.stats' % datetime.now().isoformat()
-        # can't close the stream even in a context manager: it'll be needed
-        # after the return from this method, we'll let Python's GC do its job
-        return request.make_response(
-            stream,
-            headers=[
-                ('Content-Disposition',
-                 'attachment; filename="%s"' % filename),
-                ('Content-Type', 'application/octet-stream')
-            ], cookies={'fileToken': token})
+        with tools.osutil.tempdir() as dump_dir:
+            ts = datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S")
+            filename = 'openerp_%s' % ts
+            stats_path = os.path.join(dump_dir, '%s.stats' % filename)
+            core.profile.dump_stats(stats_path)
+            pstats_list = get_pstats_print2list(
+                stats_path, sort='cumulative', limit=45,
+                exclude_fnames=[
+                    '/.repo_requirements', '/root/odoo-10.0', '/usr/', '>'])
```

Comment:
Use `~` and use `os.path.expanduser` and `os.path.expandvars` (because we could use `/home/odoo` user.